PWGLF/Tasks/cascqaanalysis.cxx

```diff
@@ -140,44 +165,155 @@ struct cascqaanalysis {
   TRandom* fRand = new TRandom();
 
   Filter preFilter =
-    nabs(aod::cascdata::dcapostopv) > dcapostopv&& nabs(aod::cascdata::dcanegtopv) > dcanegtopv&& nabs(aod::cascdata::dcabachtopv) > dcabachtopv&& aod::cascdata::dcaV0daughters < dcav0dau&& aod::cascdata::dcacascdaughters < dcacascdau;
+    (nabs(aod::cascdata::dcapostopv) > dcapostopv &&
+     nabs(aod::cascdata::dcanegtopv) > dcanegtopv &&
+     nabs(aod::cascdata::dcabachtopv) > dcabachtopv &&
+     aod::cascdata::dcaV0daughters < dcav0dau &&
+     aod::cascdata::dcacascdaughters < dcacascdau);
+
+  template <class TCascTracksTo, typename TCascade>
+  bool AcceptCascCandidate(TCascade const& cascCand, float const& pvx, float const& pvy, float const& pvz)
+  {
+    registry.fill(HIST("hCandidateCounter"), 0.5); // all candidates
+    // Access daughter tracks
+    auto v0index = cascCand.template v0_as<o2::aod::V0sLinked>();
+    auto v0 = v0index.v0Data();
+    auto posdau = v0.template posTrack_as<TCascTracksTo>();
+    auto negdau = v0.template negTrack_as<TCascTracksTo>();
+    auto bachelor = cascCand.template bachelor_as<TCascTracksTo>();
+
+    // Basic set of selections
+    if (cascCand.cascradius() > cascradius &&
+        v0.v0radius() > v0radius &&
+        cascCand.casccosPA(pvx, pvy, pvz) > casccospa &&
+        cascCand.v0cosPA(pvx, pvy, pvz) > v0cospa &&
+        TMath::Abs(posdau.eta()) < etadau &&
+        TMath::Abs(negdau.eta()) < etadau &&
+        TMath::Abs(bachelor.eta()) < etadau) {
+      return true;
+    } else {
+      return false;
+    }
+  }
 
-  void process(soa::Join<aod::Collisions, aod::EvSels, aod::Mults, aod::CentFT0Ms, aod::CentFV0As>::iterator const& collision, soa::Filtered<aod::CascDataExt> const& Cascades, aod::V0sLinked const&, aod::V0Datas const&, DauTracks const& tracks)
+  template <typename TCollision>
+  bool AcceptEvent(TCollision const& coll, bool isFillEventSelectionQA)
   {
-    // Event selection
-    if (sel8 && !collision.sel8()) {
```

Comment:
Hi @nepeivodaRS I would keep the same notation to keep changes to the bare bone

Reply:
Hi @njacazio you can merge I have no comments

Reply:
let's merge then!